notebooks/Composer_Functional.ipynb

```diff
@@ -0,0 +1,306 @@
+    "        x = torch.flatten(self.pool(x), 1)\n",
+    "        x = F.relu(self.fc1(x))\n",
+    "        x = F.relu(self.fc2(x))\n",
+    "        x = self.fc3(x)\n",
+    "        return x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, let's write a simple training loop that prints the accuracy on the test set at the end of each epoch. We'll just run a few epochs for brevity."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tqdm.notebook import tqdm\n",
+    "from composer import functional as cf\n",
+    "\n",
+    "num_epochs = 5\n",
+    "\n",
+    "def train_and_eval(model, train_loader, test_loader, device='cuda'):\n",
+    "  model = model.to(device)\n",
+    "  opt = torch.optim.Adam(model.parameters())\n",
+    "  for epoch in range(num_epochs):\n",
+    "      print(\"---- Begginning epoch \", epoch)\n",
+    "      model.train()\n",
+    "      progress_bar = tqdm(train_loader)\n",
+    "      for X, y in progress_bar:\n",
+    "          X = X.to(device)\n",
+    "          y_hat = model(X).to('cpu')\n",
+    "          loss = F.cross_entropy(y_hat, y)\n",
+    "          progress_bar.set_postfix_str(\"train loss: {:.4f}\".format(loss.detach().numpy()))\n",
+    "          loss.backward()\n",
+    "          opt.step()\n",
+    "          opt.zero_grad()\n",
+    "      model.eval()\n",
+    "      num_right = 0\n",
+    "      eval_size = 0\n",
+    "      for X, y in test_loader:\n",
+    "          y_hat = model(X.to(device)).to('cpu')\n",
+    "          num_right += (y_hat.argmax(dim=1) == y).sum().numpy()\n",
+    "          eval_size += len(y)\n",
+    "      acc_percent = 100 * num_right / eval_size\n",
+    "      print(\"Epoch {} validation accuracy: {:.2f}%\".format(epoch, acc_percent))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great. Now let's instantiate this baseline model and see how it fares on our dataset.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "baseline_model = Net()\n",
+    "baseline_trainloader, baseline_testloader = make_dataloaders()\n",
+    "train_and_eval(baseline_model, baseline_trainloader, baseline_testloader)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have this baseline, let's add algorithms to improve our data pipeline and model. We'll start by adding some data augmentation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Sequence\n",
+    "import composer.functional as cf\n",
+    "import functools\n",
+    "import sys\n",
+    "import composer\n",
+    "\n",
+    "def make_augmented_dataloaders(which_transforms: Sequence[str]):\n",
```

Comment:
I dont think we need `which_transforms`, and can just hardcode it since this is a notebook, and not meant for production configuratable code.

Reply:
also, since this is only used once.. un-function-ify?

Reply:
done